cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/FireAndForgetControllerFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.runtime.distributed.remote;
+
+import io.cdap.cdap.app.runtime.ProgramOptions;
+import io.cdap.cdap.proto.id.ProgramRunId;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * TwillControllerFactory implementation that is used by {@link FireAndForgetTwillRunnerService}.
+ */
+public class FireAndForgetControllerFactory extends ControllerFactory {
+
+  private final RemoteExecutionTwillRunnerService twillRunnerService;
+
+  FireAndForgetControllerFactory(RemoteExecutionTwillRunnerService twillRunnerService, ProgramRunId programRunId,
+                                 ProgramOptions programOpts) {
+    super(twillRunnerService, programRunId, programOpts);
+    this.twillRunnerService = twillRunnerService;
+  }
+
+  RemoteExecutionTwillController createController(ProgramRunId programRunId, ProgramOptions programOpts,
+                                                  RemoteProcessController processController,
+                                                  CompletableFuture<Void> startupTaskCompletion)
+    throws Exception {
+    RemoteExecutionService remoteExecutionService = createRemoteExecutionService(programRunId, programOpts,
```

Comment:
this is duplicated in both implementations, would be better to refactor so that it's passed into the method.

Reply:
actually, seems like the only thing that we really need to be different is what happens after the controller is created, with the normal one starting the RemoveExecutionService and attaching a listener to the controller, and this one calling release() so nothing gets monitored.

I think it would be simpler and also more clear what is happening if we added a new `monitorController(RemoteExecutionService, RemoveExecutionTwillController)` method or something like that to the base class, with this one overriding it to  just release the controller.

Reply:
This is indeed a great suggestion which results in much cleaner code. 
I updated the PR according to your suggestion. 
I though believe at some point we have to break up RemoteExecutionTwillRunnerService class as it's become too big doing so many different things.